components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoPermissionsMigration.kt

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.permission
+
+import android.content.Context
+import mozilla.components.concept.engine.permission.PermissionsMigration
+import mozilla.components.concept.engine.permission.SitePermissionsStorage
+
+
+class GeckoPermissionsMigration(
+    private val context: Context,
+    private val sourceStorage: SitePermissionsStorage,
+    private val targetStorage: GeckoSitePermissionsStorage
+) : PermissionsMigration {
+
+
+    override fun isFinished(): Boolean {
+        // TODO: A boolean flag backed by SharedPreferences
+        return false
+    }
+
+    override fun performMigration() {
```

Comment:
The main idea is to call `performMigration` on Fenix startup, this will take some time depending on the amount of user data but it will be done once, also we are going to block when a permission request arrive until the migration is finished.

Reply:
I think it will be much easier if we only have one blocking migration, instead doing `async` and have to synchronized multiple migrations at the time.